### PR TITLE
MINOR [Go] Add nested struct test for pqarrow

### PR DESCRIPTION
**Pull request by candiduslynx**
@zeroshade I've found a disturbing behavior with structs (namely, nested ones) in pqarrow.

This blocks https://github.com/cloudquery/filetypes/pull/172

Could you please take a look?

### CODE REVIEW

File: go/parquet/pqarrow/encode_arrow_test.go [hunk 1]
```diff
@@ -931,6 +931,64 @@ func (ps *ParquetIOTestSuite) TestReadDecimals() {
...
+	rgw := writer.AppendRowGroup()
+	cw, err := rgw.NextColumn()
+	ps.NoError(err)
+
+	props := pqarrow.NewArrowWriterProperties(pqarrow.WithAllocator(mem))
+	ctx := pqarrow.NewArrowWriteContext(context.TODO(), &props)
+	ps.NoError(pqarrow.WriteArrowToColumn(ctx, cw, expected, nil, nil, false))
+	ps.NoError(cw.Close())
+	ps.NoError(rgw.Close())
+	ps.NoError(writer.Close())
```
Comment (zeroshade) [inline go/parquet/pqarrow/encode_arrow_test.go:978]: I'm extremely confused here, you're writing a nested struct arrow array to a column expecting a single leaf bytearray of decimal data.....? It feels like this should error at a minimum. There's no way this should work, unless I'm missing something.

**Comment by candiduslynx** [comment]
@zeroshade I found an issue with the test & it should now pass, but I'm really confused about `array.NewStructData` behavior in conjunction with pqarrow.
Could you please take a look at the code in https://github.com/cloudquery/filetypes/pull/172?
Specifically, the issue I see is that the offsets/null bitmaps for the nested field are propagated strangely.
It [panics](https://github.com/cloudquery/filetypes/actions/runs/5134969218/jobs/9239730687?pr=172#step:6:8) while trying to create Struct array from data, so I don't know where to look.

**Comment by candiduslynx** [comment]
I guess the issue can be connected to https://github.com/apache/arrow/blob/main/go/parquet/pqarrow/encode_arrow_test.go#L1040-L1042
```
// current impl of ArrayEquals for structs doesn't correctly handle nulls in the parent
// with a non-nullable child when comparing. Since after the round trip, the data in the
// child will have the nulls, not the original data.
```

**Comment by candiduslynx** [comment]
I'm closing this for now, however, it would've been nice to be able to use `array.NewStructData` on the read values

**Pull request closed without merging**